File: day4/Ruby/partB_sol.rb
=begin
@author: aaditkamat
@date: 25/12/2018
=end
def most_frequent_character(str)
	if str === nil or not str.class === String
		nil
	end
	counts = {}
	str.each_char do |ch|
		if not counts.key?(ch)
			counts[ch] = 1
		else
			counts[ch] = counts[ch] + 1
		end
	end
	counts.key(counts.values.max)
end

print "Enter a string: "
str = gets
str.chomp!
puts "The most frequent character in #{str} is : #{most_frequent_character(str)}"

File: day4/Python/partB_sol.py
'''
        * @author: prateek3255
        * @date: 25/12/18
'''

string=input("Enter the string : ")
characters={}

for char in string:
    if char.lower() in characters:
        characters[char.lower()]+=1
    else:
        characters[char.lower()]=1
        
print("The most occouring character in the string is : ", max(characters,key=characters.get))
